Userland/Libraries/LibWeb/CSS/StyleValue.cpp

```diff
@@ -1642,6 +1648,19 @@ bool PositionStyleValue::equals(StyleValue const& other) const
         && m_offset_y == typed_other.m_offset_y;
 }
 
+String RectStyleValue::to_string() const
+{
+    return String::formatted("top_edge: {}, right_edge: {}, bottom_edge: {}, left_edge: {}", m_rect.top_edge, m_rect.right_edge, m_rect.bottom_edge, m_rect.left_edge);
+}
+
+bool RectStyleValue::equals(StyleValue const& other) const
+{
+    if (type() != other.type())
+        return false;
+    auto const& typed_other = other.as_rect();
+    return m_rect.top_edge == typed_other.rect().top_edge && m_rect.right_edge == typed_other.rect().right_edge && m_rect.bottom_edge == typed_other.rect().bottom_edge && m_rect.left_edge == typed_other.rect().left_edge;
```

Comment:
I think this would be slightly nicer if you define a `operator==` for `EdgeRect` first then just do `m_rect() == other.as_rect().m_rect`. (See how this is done for LinearGradients)